cloud-native/boot-loader/src/main/kotlin/com/linecorp/centraldogma/config/AuthConfig.kt

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.config
+
+import com.google.common.base.Ascii
+import com.linecorp.centraldogma.internal.Jackson
+import com.linecorp.centraldogma.server.ReplicationConfig
+import com.linecorp.centraldogma.server.ReplicationMethod
+import com.linecorp.centraldogma.server.ZooKeeperReplicationConfigSpec
+import com.linecorp.centraldogma.server.ZooKeeperServerConfigSpec
+import com.linecorp.centraldogma.server.auth.AuthConfigSpec
+import com.linecorp.centraldogma.server.auth.AuthProviderFactory
+import org.springframework.validation.annotation.Validated
+import java.util.function.Function
+
+open class ReplicationConfigImpl(
+    private val method: ReplicationMethod,
+) : ReplicationConfig {
+    override fun method(): ReplicationMethod = method
+}
+
+data class NoneReplicationConfig(private val method: ReplicationMethod) : ReplicationConfigImpl(method)
+
+@Validated
+data class ZooKeeperServerConfig(
+    private val host: String,
+    private val quorumPort: Int,
+    private val electionPort: Int,
+    private val clientPort: Int,
+    private val groupId: Int?,
+    private val weight: Int = 1,
+) : ZooKeeperServerConfigSpec {
+    init {
+        check(weight >= 0) { "weight: $weight (expected: >= 0)" }
+    }
+
+    override fun host(): String = host
+
+    override fun quorumPort(): Int = ZooKeeperServerConfigSpec.validatePort(quorumPort, "quorumPort")
+
+    override fun electionPort(): Int = ZooKeeperServerConfigSpec.validatePort(electionPort, "electionPort")
+
+    override fun clientPort(): Int = clientPort
+
+    override fun groupId(): Int? = groupId
+
+    override fun weight(): Int = weight
+}
```

Comment:
Discussion point 1.

Due to hard coupling between configs and jackson deserialization, I wanna suggest to split configuration spec and it's implementation.

This approach can separate dependencies between jackson and others.